File: common/src/main/java/com/hartwig/actin/datamodel/clinical/ClinicalStatus.java
package com.hartwig.actin.datamodel.clinical;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class ClinicalStatus {

    @Nullable
    public abstract Integer who();

    @Nullable
    public abstract Boolean hasActiveInfection();

    @Nullable
    public abstract Boolean hasSigAberrationLatestEcg();

    @Nullable
    public abstract String ecgAberrationDescription();

}

File: common/src/main/java/com/hartwig/actin/clinical/datamodel/ClinicalStatus.kt
package com.hartwig.actin.clinical.datamodel

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class ClinicalStatus {
    abstract fun who(): Int?
    abstract fun infectionStatus(): InfectionStatus?
    abstract fun ecg(): ECG?
    abstract fun lvef(): Double?
    abstract fun hasComplications(): Boolean?
}
